enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/context.tsx

```diff
@@ -1,83 +1,97 @@
 import {
   createContext,
-  type PropsWithChildren,
   useContext,
+  useEffect,
+  type PropsWithChildren,
   useMemo,
 } from "react";
 
 import type { SdkPluginsConfig } from "embedding-sdk";
+import { useLoadQuestion } from "embedding-sdk/hooks/private/use-load-question";
+import type { LoadSdkQuestionParams } from "embedding-sdk/lib/load-question";
 import { useSdkSelector } from "embedding-sdk/store";
 import { getPlugins } from "embedding-sdk/store/selectors";
+import type {
+  NavigateToNewCardParams,
+  SdkQuestionResult,
+} from "embedding-sdk/types/question";
 import type { Mode } from "metabase/visualizations/click-actions/Mode";
 import { getEmbeddingMode } from "metabase/visualizations/click-actions/lib/modes";
+import type Question from "metabase-lib/v1/Question";
 
-import {
-  type UseLoadQuestionParams,
-  useInteractiveQuestionData,
-  useLoadQuestion,
-} from "./hooks";
-
-type InteractiveQuestionContextType = {
+interface InteractiveQuestionContextType extends SdkQuestionResult {
   plugins: SdkPluginsConfig | null;
   mode: Mode | null | undefined;
   isQuestionLoading: boolean;
   resetQuestion: () => void;
   onReset?: () => void;
   onNavigateBack?: () => void;
-};
+  onQuestionChange: (question: Question) => void;
+  onNavigateToNewCard: (params: NavigateToNewCardParams) => void;
+}
 
 /**
  * Note: This context should only be used as a wrapper for the InteractiveQuestionResult
  * component. The idea behind this context is to allow the InteractiveQuestionResult component
  * to use components within the ./components folder, which use the context for display
- * and functions. Any data that can be referenced from the store should be placed in
- * the useInteractiveQuestionData hook.
+ * and functions.
  * */
 export const InteractiveQuestionContext = createContext<
   InteractiveQuestionContextType | undefined
 >(undefined);
 
 type InteractiveQuestionProviderProps = PropsWithChildren<
-  UseLoadQuestionParams & {
+  {
     componentPlugins?: SdkPluginsConfig;
     onReset?: () => void;
     onNavigateBack?: () => void;
-  }
+  } & LoadSdkQuestionParams
 >;
 
 export const InteractiveQuestionProvider = ({
-  children,
   location,
   params,
   componentPlugins,
   onReset,
   onNavigateBack,
+  children,
 }: InteractiveQuestionProviderProps) => {
-  const { isQuestionLoading, resetQuestion } = useLoadQuestion({
-    location: location,
-    params: params,
-  });
+  const {
+    question,
+    queryResults,
+    isQuestionLoading,
```

Comment:
We now rely on React context instead of Redux for the multiple interactive questions, so it makes sense to use `useInteractiveQuestionContext` as a source-of-truth here.

Reply:
@oisincoveney looks like the reloading issue is fixed after adding a separate `isQueryRunning` state - it only shows the loading indicator in the visualizations. usability-wise it seems quite smooth.

<div>
    <a href="https://www.loom.com/share/1a388cf2d4b245aba75568d613dddf34">
      <p>Multiple Interactive Questions Demo - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/1a388cf2d4b245aba75568d613dddf34">
      <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/1a388cf2d4b245aba75568d613dddf34-with-play.gif">
    </a>
  </div>

Reply:
Nice! Looks good then :smile: